lib/core/EventBus.js

```diff
@@ -332,9 +332,14 @@ EventBus.prototype._invokeListeners = function(event, args, listeners) {
 
   var idx,
       listener,
-      returnValue;
+      returnValue,
+      listenersCopy;
+
+  // make a shallow copy of the listeners
+  // due to possible mutation of the listeners array
+  listenersCopy = listeners.slice(0);
```

Comment:
Creating a shallow copy may cause quite a performance penalty. A better solution could be to keep the listener index an instance property and adjust it once removing `only` listeners.

Reply:
Would something like this be better if slice is possibly too slow
``` 

  for (idx = 0; (listener = listeners[idx]);) {

    // handle stopped propagation
    if (event.cancelBubble) {
      break;
    }
    var oldLength = listeners.length;
    returnValue = this._invokeListener(event, args, listener);
    if (oldLength === listeners.length) {
      idx++;
    }
  }